runtime/gc_glue_java/HeadlessMixedObjectScanner.hpp

```diff
@@ -99,7 +96,13 @@ class GC_HeadlessMixedObjectScanner : public GC_ObjectScanner
 #endif /* J9VM_GC_LEAF_BITS */
 		}
 	}
-	
+
+	MMINLINE void
+	initialize(MM_EnvironmentBase *env, uintptr_t *descriptionPtr)
+	{
+		initialize(env, descriptionPtr, NULL);
+	}
+
```

Comment:
May be it is better way to organize this `J9VM_GC_LEAF_BITS` support for `initialize ()`. Will discuss later on, good for now

Reply:
I think this is horrible as well, and I cringed writing this code. The best alternative I can think of is guarding the signature:
```
#if defined(J9VM_GC_LEAF_BITS)
	initialize(MM_EnvironmentBase *env, uintptr_t *descriptionPtr, uintptr_t *leafPtr)
#else /* J9VM_GC_LEAF_BITS */
	initialize(MM_EnvironmentBase *env, uintptr_t *descriptionPtr)
#endif /* J9VM_GC_LEAF_BITS */
{
```

Reply:
We can have two versions of function exactly as for `getNextSlotMap()`. I am not sure it is much better (need to modify two places instead of one in case) bur easier to read

Reply:
Bigger question is why we maintain both variants. Anybody aware of drawbacks of having 'leaf' functionality? I'm not. 

I'd consider removing non-leaf variant across OMR (or at least across OpenJ9),. But of course, it's a change for itself outside of this scope.

Reply:
Well I believe it was an assumption that another language would use bit array to mark object reference slots but at the same time does not support leaves optimization obviously. I agree non-leaf version should be removed, it is not practical from my point of view.

Reply:
I would be all for removing it.  It would let us get rid of some code duplication.

Reply:
Are we okay with leaving this as is for now?